core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -897,30 +901,15 @@ object GraphQLActionBuilder {
                 unionTypes,
                 interfaceTypes,
                 accum,
-                maxNumberOfGenes
+                maxNumberOfGenes,
+                tableFieldWithArgs
             )
 
             //Remove primitive types (scalar and enum) from return params
-            if (gene.name.lowercase() != "scalar"
-                && !(gene is OptionalGene && gene.gene.name == "scalar")
-                && !(gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is OptionalGene && gene.gene.template.name.lowercase() == "scalar")
-                && !(gene is ArrayGene<*> && gene.template.name.lowercase() == "scalar")
-                && !(gene is ArrayGene<*> && gene.template is OptionalGene && gene.template.name.lowercase() == "scalar")
-                && !(gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template.name.lowercase() == "scalar")
-                //enum cases
-                && !(gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is OptionalGene && gene.gene.template.gene is EnumGene<*>)
-                && !(gene is ArrayGene<*> && gene.template is EnumGene<*>)
-                && !(gene is ArrayGene<*> && gene.template is OptionalGene && gene.template.gene is EnumGene<*>)
-                && !(gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is EnumGene<*>)
-                && !(gene is EnumGene<*>)
-                && !(gene is OptionalGene && gene.gene is EnumGene<*>)
-
-            ) {
-                params.add(GQReturnParam(methodName, gene))
-            }
+            if (isReturnPrimitive(gene)) params.add(GQReturnParam(methodName, gene))
```

Comment:
is the comment above correct? looks like you are adding it when a primitive

Reply:
done